Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -453,266 +453,9 @@ u32 HostRead_Instruction(const u32 address)
-	// {0x804D7720, 0x4},     // [804D7720 - 804D7724)
-	// {0x804D7744, 0x4},     // [804D7744 - 804D7748)
-	// {0x804D774C, 0x8},     // [804D774C - 804D7754)
-	// {0x804D7758, 0x8},     // [804D7758 - 804D7760)
-	// {0x804D7788, 0x10},    // [804D7788 - 804D7798)
-	// {0x804D77C8, 0x4},     // [804D77C8 - 804D77CC)
-	// {0x804D77D0, 0x4},     // [804D77D0 - 804D77D4)
-	// {0x804D77E0, 0x4},     // [804D77E0 - 804D77E4)
-	// {0x804DE358, 0x80},    // [804DE358 - 804DE3D8)
-	// {0x804DE800, 0x70},    // [804DE800 - 804DE870)
-	//};
-
-	// auto sceneController = ReadFromHardware<FLAG_READ, u32>(0x80479d30);
-	// if ((sceneController & 0xFF0000FF) != 0x08000002)
-	//{
-	// return;
-	//}
-
-	// auto isLoading = ReadFromHardware<FLAG_READ, u32>(0x80479d64);
-	// if (isLoading)
-	//{
-	// return;
-	//}
-
-	// if (address >= 0x804dec00)
-	//{
-	// return;
-	//}
-	////if (!write)
-	////{
-	////	return;
-	////}
-
-	// if (visited.count(address))
-	//{
-	// return;
-	//}
-
-	// visited[address] = true;
-
-	// for (auto it = soundStuff.begin(); it != soundStuff.end(); ++it)
-	//{
-	// if (address >= it->address && address < it->address + it->length)
-	// {
-	//     return;
-	// }
-	//}
-
-	// if ((address & 0xFF000000) == 0xcc000000)
-	//{
-	// return;
-	//}
-
-	// std::vector<Dolphin_Debugger::CallstackEntry> callstack;
-	// Dolphin_Debugger::GetCallstack(callstack);
-
-	// bool isFound = false;
-	// for (auto it = callstack.begin(); it != callstack.end(); ++it)
-	//{
-	// std::string func = PowerPC::debug_interface.GetDescription(it->vAddress).c_str();
-	// if (whitelist.count(func))
-	// {
-	//     isFound = true;
-	//     break;
-	// }
-	//}
-
-	// if (!isFound)
-	//{
-	// return;
-	//}
-
-	// NOTICE_LOG(MEMMAP, "(%s) %x (%s) | %x (%x) <-> %x", write ? "Write" : "Read", PC,
-	//        PowerPC::debug_interface.GetDescription(PC).c_str(), var, size, address);
-
-	//*********************************************************************
-	//* Detect writes in unknown memory region
-	//*********************************************************************
-	//static std::unordered_map<u32, bool> visited = {};
-
-	//auto sceneController = ReadFromHardware<FLAG_READ, u32>(0x80479d30);
-	//if ((sceneController & 0xFF0000FF) != 0x08000002)
-	//{
-	//	return;
-	//}
-
-	//auto isLoading = ReadFromHardware<FLAG_READ, u32>(0x80479d64);
-	//if (isLoading)
-	//{
-	//	return;
-	//}
-
-	//if (!write)
-	//{
-	//	return;
-	//}
-
-	//// [804fec00 - 80BD5C40)
-	//if (address < 0x8071b000 || address >= 0x80bb0000)
-	//{
-	//	return;
-	//}
-
-	//if (visited.count(address))
-	//{
-	//	return;
-	//}
-
-	//visited[address] = true;
-
-	//if ((address & 0xFF000000) == 0xcc000000)
-	//{
-	//	return;
-	//}
-
-	//NOTICE_LOG(MEMMAP, "(%s) %x (%s) | %x (%x) <-> %x", write ? "Write" : "Read", PC,
-	//           PowerPC::debug_interface.GetDescription(PC).c_str(), var, size, address);
+	if (!PowerPC::memchecks.HasAny())
+		return;
 
-#ifdef ENABLE_MEM_CHECK
```

Comment:
Curious why you deleted this ifdef?

Reply:
That is actually a missing commit ....
hold on